pathd/path_pcep_controller.c

```diff
@@ -673,20 +752,54 @@ int pcep_thread_event_update_pcc_options(struct ctrl_state *ctrl_state,
 	ctrl_state->pcc_opts = opts;
 	return 0;
 }
-
+int pcep_thread_event_update_best_pce(struct ctrl_state *ctrl_state, int best)
+{
+	PCEP_DEBUG(" recalculating pce precedence ");
+	if (best) {
+		struct pcc_state *best_pcc_state =
+			get_pcc_state(ctrl_state, best);
+		if (best_pcc_state->previous_best != best_pcc_state->is_best) {
+			PCEP_DEBUG(" %s Resynchro best (%i) previous best (%i)",
+				   best_pcc_state->tag, best_pcc_state->id,
+				   best_pcc_state->previous_best);
+			best_pcc_state->status = PCEP_PCC_SYNCHRONIZING;
```

Comment:
We shouldn't be reading pcc state structure fields, but we should DEFINITELY NOT be modifying them. if there is a good reason, then add a function in the pcc to do what makes sens to do there.

Reply:
moved to a function